cypress/e2e/addrule.cy.ts

```diff
@@ -0,0 +1,47 @@
+describe('Query Builder', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000');
+    cy.get('[data-testid="btn-add-rule"]').click();
+  });
+
+  it('user can add rule by clicking the ADD RULE button', () => {
+    cy.get('[data-testid="rule-item"]').should('exist');
+    cy.get('[data-testid="field-select"]')
+      .select('Age')
+      .should('have.value', 'Age');
+
+    cy.get('[data-testid="operator-select"]')
+      .select('>')
+      .should('have.value', '>');
+
+    cy.get('[data-testid="general-input"]')
+      .type('10')
+      .should('have.value', '10');
+
+    cy.get('[data-testid="query-output"]')
```

Comment:
Why are you invoking this? I thought the text appears in the output box as one types? So ideally we would select the element that contains the query output and assert that the text we expect is there? Or am I confused?

Reply:
I am invoking it in order to extract the specific text I am asserting against. The element returns the entire content of the JSON

Reply:
Oh, I hadn't seen that you were getting the text from the element, I thought you had created a separate object. Okay makes sense!